src/labone/mock/demo_mock.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+from pathlib import Path
+from typing import Any, Coroutine
+from labone.core import ListNodesFlags
+from labone.core.helper import LabOneNodePath
+
+from labone.core.reflection import ReflectionServer
+from labone.core.session import Session
+from labone.core.subscription import StreamingHandle
+from labone.core.value import AnnotatedValue
+from labone.mock.automatic_hpk_functionality import AutomaticHpkFunctionality
+from labone.mock.hpk_mock_server import HpkMockServer
+from labone.mock.hpk_functionality import HpkMockFunctionality
+from labone.mock.mock_server import MockServer
+
+
+class TestHPK(AutomaticHpkFunctionality):
+
+    
+    def set_with_expression(self, value: AnnotatedValue) -> Coroutine[Any, Any, list[AnnotatedValue]]:
+        raise ValueError("(test) e.g. tried to set invalid path")
+
+    def get_with_expression(self, path_expression: LabOneNodePath, flags: ListNodesFlags | int = ListNodesFlags.ABSOLUTE | ListNodesFlags.RECURSIVE | ListNodesFlags.LEAVES_ONLY | ListNodesFlags.EXCLUDE_STREAMING | ListNodesFlags.GET_ONLY) -> Coroutine[Any, Any, list[AnnotatedValue]]:
+        raise ValueError("(test) e.g. tried to set invalid path")
+    pass
+
+async def main():
+    mock_server = MockServer(
+        capability_bytes=Path("test.bin"),
+        concrete_server=HpkMockServer(
+            AutomaticHpkFunctionality(
+                paths_to_info={
+                    "/a/b/c": {"Description": "some path"},
+                    "/a/x/y": {"Properties": "Read, Write"},
+                    "/a/x/z/q": {},
+                }
+            )
+        ),
+    )
```

Comment:
We should add a helper function to `labone.mock` that does this. e.g. `spawn_hpk_mock(paths_to_info)`

Reply:
I would even make the exposed functionality directly use `AutomaticHpkFunctionality` and accept `paths_to_info`

Reply:
But this wouldn't be in the spirit of subclassing `AutomaticHpkFunctionality`

Reply:
Alright, but we need to make an example in the docstring of this function how to use it with the AutomaticHpkFunctionality ... because most customers will use this mock without deriving or overwriting anything